website/blog/2023-04-18-building-a-kimball-dimensional-model-with-dbt.md

```diff
@@ -0,0 +1,712 @@
+    select * from {{ ref('dim_order_status') }}
+),
+
+d_product as (
+    select * from {{ ref('dim_product') }}
+),
+
+d_date as (
+    select * from {{ ref('dim_date') }}
+)
+
+select
+    {{ dbt_utils.star(from=ref('fct_sales'), relation_alias='f_sales', except=[
+        "product_key", "customer_key", "creditcard_key", "ship_address_key", "order_status_key", "order_date_key"
+    ]) }},
+    {{ dbt_utils.star(from=ref('dim_product'), relation_alias='d_product', except=["product_key"]) }},
+    {{ dbt_utils.star(from=ref('dim_customer'), relation_alias='d_customer', except=["customer_key"]) }},
+    {{ dbt_utils.star(from=ref('dim_credit_card'), relation_alias='d_credit_card', except=["creditcard_key"]) }},
+    {{ dbt_utils.star(from=ref('dim_address'), relation_alias='d_address', except=["address_key"]) }},
+    {{ dbt_utils.star(from=ref('dim_order_status'), relation_alias='d_order_status', except=["order_status_key"]) }},
+    {{ dbt_utils.star(from=ref('dim_date'), relation_alias='d_date', except=["date_key"]) }}
+from f_sales
+left join d_product on f_sales.product_key = d_product.product_key
+left join d_customer on f_sales.customer_key = d_customer.customer_key
+left join d_credit_card on f_sales.creditcard_key = d_credit_card.creditcard_key
+left join d_address on f_sales.ship_address_key = d_address.address_key
+left join d_order_status on f_sales.order_status_key = d_order_status.order_status_key
+left join d_date on f_sales.order_date_key = d_date.date_key
+```
+
+In the OBT above, we perform joins between the fact and dimension tables using the surrogate keys. 
+
+Using `dbt_utils.star()`, we select all columns except the surrogate key columns since the surrogate keys don't hold any meaning besides being useful for the joins. 
+
+We can then build the OBT by running `dbt run`. Your dbt DAG should now look like this: 
+
+<Lightbox src="/img/blog/2023-04-18-building-a-kimball-dimensional-model-with-dbt/dbt-dag.png" width="85%" title="Final dbt DAG"/>
+
+Congratulations, you have reached the end of this tutorial. If you want to learn more, please see the learning resources below on dimensional modelling. 
+
+## Learning resources
+
+- [Kimball group learning resources](https://www.notion.so/Building-a-Kimball-dimensional-model-with-dbt-7fc6e04e26154c7baf67f695ae2a1ae2)
+- [The Data Warehouse toolkit book](https://www.kimballgroup.com/data-warehouse-business-intelligence-resources/books/data-warehouse-dw-toolkit/)
+- [Dbt discourse on whether dimensional modelling is still relevant](https://discourse.getdbt.com/t/is-kimball-dimensional-modeling-still-relevant-in-a-modern-data-warehouse/225)
+- [Dbt glossary on dimensional modelling](https://docs.getdbt.com/terms/dimensional-modeling)
+
+## Let's talk 
+
+The materials in this article were created by [Data Engineer Camp](https://dataengineercamp.com/), a 16-week data engineering bootcamp for professionals looking to transition to data engineering and analytics engineering.
```

Comment:
Hey @jonathanneo this is amazing!! Do you mind moving this paragraph under your author's note below? Thanks!

Reply:
Hey @KiraFuruichi this is now done :)